rest/rest-resources/src/main/java/org/apache/brooklyn/rest/security/provider/LdapSecurityProvider.java

```diff
@@ -99,13 +108,76 @@ public boolean authenticate(HttpServletRequest request, Supplier<HttpSession> se
             env.put(Context.SECURITY_PRINCIPAL, getSecurityPrincipal(user));
             env.put(Context.SECURITY_CREDENTIALS, pass);
 
-            new InitialDirContext(env);  // will throw if password is invalid
+            DirContext ctx = new InitialDirContext(env);// will throw if password is invalid
```

Comment:
Feels like this needs a try finally block?

Reply:
I don't think it's needed. If the authentication fails an ti throw an exception, the context won't be created and the user remains unathenticated.
Why do you think is needed @duncangrant ?